lib/model/partita.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:alley_app/model/evento.dart';
+
+class Partita extends Evento {
+  SquadraPartecipante casa;
+  SquadraPartecipante ospite;
+
+  Partita({uid, dataEOra, luogo, this.casa, this.ospite}) : super(uid: uid, dataEOra: dataEOra, luogo: luogo);
+
+  static Partita fromJson(Map<String, dynamic> json) {
+    return Partita(
+      uid: json['uid'],
+      dataEOra: json['dataEOra'].toDate(),
+      luogo: json['luogo'],
+      casa: SquadraPartecipante.fromJson(json['casa']),
+      ospite: SquadraPartecipante.fromJson(json['ospite']),
+    );
+  }
+
+  @override
+  String toString() {
+    return '$dataEOra a $luogo';
+  }
+}
+
+class SquadraPartecipante {
+  String idSquadra;
+  String nome;
+  int punti;
+  String turnoCibo;
+  List<Convocazione> convocazioni;
+
+  SquadraPartecipante({this.idSquadra, this.nome, this.punti, this.turnoCibo, this.convocazioni});
+
+  static SquadraPartecipante fromJson(Map<String, dynamic> json) {
```

Comment:
Quando ricostruiamo gli oggetti dal json (`Map<String, dynamic>` come parametro) a volte lo chiamiamo `fromJson`, altre volte (vedi in partite_db.dart) lo chiamiamo `_buildSquadra`, altre volte ancora lo chiamiamo `fromSnapshot`. Non sarebbe meglio trovare una linea comune?

Reply:
fromJson è per le `Map<String, dynamic>`, fromSnaphot da una `QuerySnapshot`. Si possono comunque ricondurre uno all'altro. La `_buildSquadra` è stata fatta perchè la sua fromJson esplode